pandas/tests/generic/test_generic.py

```diff
@@ -612,6 +612,16 @@ def test_pct_change(self, periods, fill_method, limit, exp):
         else:
             tm.assert_series_equal(res, Series(exp))
 
+    @pytest.mark.parametrize("data,index,expected_first,expected_last", [
+        ({'A': [1, 2, 3]}, [1, 1, 2], 1, 2),
+        ({'A': [1, 2, 3]}, [1, 2, 2], 1, 2),
+        ({'A': [1, 2, 3, 4]}, ['d', 'd', 'd', 'd'], 'd', 'd')])
+    def test_valid_index(self, data, index, expected_first, expected_last):
```

Comment:
do we not already have *some* tests for this? pls put near the others. does this duplicate existing tests at all?

Reply:
> do we not already have some tests for this? pls put near the others.

Thanks - The only test involving `first_valid_index` and `last_valid_index` is  in `./pandas/tests/frame/test_timeseries.py` - and does not specifically check for duplicate
first or last index values. Would you suggest I move this test there?